nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RetryFlowFile.java

```diff
@@ -0,0 +1,314 @@
+    private Set<Relationship> relationships;
+    private String retryAttribute;
+    private Boolean penalizeRetried;
+    private Boolean failOnOverwrite;
+    private String reuseMode;
+    private String lastRetriedBy;
+
+    public static final PropertyDescriptor RETRY_ATTRIBUTE = new PropertyDescriptor.Builder()
+            .name("retry-attribute")
+            .displayName("Retry Attribute")
+            .description("The name of the attribute that contains the current retry count for the FlowFile. " +
+                    "WARNING: If the name matches an attribute already on the FlowFile that does not contain a " +
+                    "numerical value, the processor will either overwrite that attribute with '1' or fail " +
+                    "based on configuration.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_EL_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .defaultValue("flowfile.retries")
+            .build();
+    public static final PropertyDescriptor MAXIMUM_RETRIES = new PropertyDescriptor.Builder()
+            .name("maximum-retries")
+            .displayName("Maximum Retries")
+            .description("The maximum number of times a FlowFile can be retried before being " +
+                    "passed to the 'retries_exceeded' relationship")
+            .required(true)
+            .addValidator(StandardValidators.createLongValidator(1, Integer.MAX_VALUE, true))
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .defaultValue("3")
+            .build();
+    public static final PropertyDescriptor PENALIZE_RETRIED = new PropertyDescriptor.Builder()
+            .name("penalize-retries")
+            .displayName("Penalize Retries")
+            .description("If set to 'true', this Processor will penalize input FlowFiles before passing them " +
+                    "to the 'retry' relationship. This does not apply to the 'retries_exceeded' relationship.")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .build();
+    public static final PropertyDescriptor FAIL_ON_OVERWRITE = new PropertyDescriptor.Builder()
+            .name("Fail on Non-numerical Overwrite")
+            .description("If the FlowFile already has the attribute defined in 'Retry Attribute' that is " +
+                    "*not* a number, fail the FlowFile instead of resetting that value to '1'")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .build();
+
+    public static final AllowableValue FAIL_ON_REUSE = new AllowableValue(
+            "fail",
+            "Fail on Reuse",
+            "If a FlowFile's UUID does not match the supplied retry UUID, fail the FlowFile" +
+                    " regardless of current retry count"
```

Comment:
> If a FlowFile's UUID does not match ...

should be

> If this processor's UUID does not match ...

Is that correct? The same applies to `WARN_ON_REUSE` and `RESET_ON_REUSE`

Reply:
You are correct, hopefully the new wording is clear